### ISSUE TITLE
Cannot query "WHERE 1 = 0" with ORDER BY clause

### ISSUE THREAD

**Issue by dphil** [dolthub/dolt#3972]
It may seem like an odd query, but is what knex produces when you use a WHERE IN clause where you are checking for membership in an empty array. For reference see https://github.com/knex/knex/issues/2980#issuecomment-539535674

To reproduce (tested with dolt 0.40.20):
```
mysql> CREATE TABLE `bug_test` (`name` varchar(255), PRIMARY KEY (`name`));
mysql> select * from bug_test where 1 = 0; -- works fine
mysql> select * from bug_test where 1 = 0 order by `name` asc; -- throws error
Error 1105: unable to push plan.Sort node below *plan.emptyTable
```

This error does not occur on mysql (tested with 8.0.29).

**Comment by timsehn** [comment]
Great bug. @max-hoffman or @fulghum you wanna fix quick?

**Comment by max-hoffman** [comment]
@dphil thanks for the bug report! I have a PR that adds a test [here](https://github.com/dolthub/go-mysql-server/pull/1140).

**Comment by timsehn** [comment]
To be clear, the bug was already fixed in the tip of main. We just added a test to make sure it did not recur.

**Comment by fulghum** [comment]
The fix for this issue was just released in [Dolt 0.40.21](https://github.com/dolthub/dolt/releases/tag/v0.40.21). Check it out when you get a chance and it should be working smoothly for you now. 

Thanks again for taking the time to report issues to us! Let us know if you hit any other issues and we'll be happy to jump on them.
